src/CDSAPI.jl

```diff
@@ -2,6 +2,50 @@ module CDSAPI
 
 using HTTP
 using JSON
+using Base.ScopedValues
+
+const auth = ScopedValue(Dict("url" => "", "key" => ""))
+
+"""
+    CDScredentials()
+
+Checks the default location for the cds credentials
```

Comment:
```suggestion
Attempt to find CDS credentials using different methods.
```

Please document these methods, particularly their precedence order.